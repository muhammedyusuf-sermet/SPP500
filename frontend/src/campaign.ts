export interface ICampaignState {
	Id?: string,
	Name: string,
	Description: string;
	// Monsters: Monster.IMonsterState[];
}
