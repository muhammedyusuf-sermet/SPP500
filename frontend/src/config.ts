export const API_URL = "https://http://3.18.65.138:3000";