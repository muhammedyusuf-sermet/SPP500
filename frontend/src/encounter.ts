import { IMonsterData } from "./monster";

export interface IEncounterData {
	Id?: string,
	Name: string,
	Description: string;
	Monsters: IMonsterData[];
}
