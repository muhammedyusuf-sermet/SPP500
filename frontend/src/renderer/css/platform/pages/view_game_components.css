.view-game-components-container {
	text-align: center;
}
.page-title {
	padding: 5px 15px;
	display: inline-block;
}

.view-game-components-container .tab-root {
	flex-grow: 1;
}

.view-game-components-container .app-bar {
	position: relative;
}

.view-game-components-container .layout {
	width: auto;
	margin-left: 2%;
	margin-right: 2%;
}

.view-game-components-container .card-grid {
	padding: 8px 0;
}

.view-game-components-container .card {
	height: 100%;
	display: flex;
	flex-direction: column;
}

.view-game-components-container .card-media {
	padding-top: 56.25%; /* 16:9 */
}

.view-game-components-container .card-content {
	flex-grow: 1;
}
