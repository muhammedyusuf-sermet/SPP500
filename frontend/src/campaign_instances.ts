import * as Campaign from "./campaign";

// Dummy array of monsters
import EncounterInstances from './encounter_instances';

let encounterExample: Campaign.ICampaignState;
encounterExample = {
	Id: 0,
	Name: "Campaign",
	Summary: "Bloody hell...",
	Notes: "You gonna die!",
	Encounters: EncounterInstances['EncounterInstances'],
};

let CampaignInstances = new Array();

// Generate a dummy array of encounters with different IDs
for (var i = 0; i < 12; i++) {
	let encounter = {} as Campaign.ICampaignState;
	Object.assign(encounter, encounterExample);
	encounter['Id'] = i;
	CampaignInstances.push(encounter);
}

export default {
    CampaignInstances,
}