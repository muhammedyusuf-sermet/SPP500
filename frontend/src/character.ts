import { ICampaignData } from "./campaign";

export enum CharacterRace {
	Dragonborn = "Dragonborn",
	Dwarf = "Dwarf",
	Elf = "Elf",
	Gnome = "Gnome",
	HalfElf = "HalfElf",
	Halfling = "Halfling",
	HalfOrc = "HalfOrc",
	Human = "Human",
	Tiefling = "Tiefling"
}

export enum CharacterClass {
	Barbarian = "Barbarian",
	Bard = "Bard",
	Cleric = "Cleric",
	Druid = "Druid",
	Fighter = "Fighter",
	Monk = "Monk",
	Paladin = "Paladin",
	Ranger = "Ranger",
	Rogue = "Rogue",
	Sorcerer = "Sorcerer",
	Warlock = "Warlock",
	Wizard = "Wizard"
}

export interface ICharacterData {
	Id?: number;
	Name: string;
	Level?: number;
	Race?: CharacterRace;
	Class?: CharacterClass;
	MaxHealth?: number;
	ArmorClass?: number;
	Notes?: string;

	Campaigns?: ICampaignData[];
}
