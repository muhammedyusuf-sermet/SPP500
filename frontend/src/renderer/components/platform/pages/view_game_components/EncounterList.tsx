import * as React from 'react';
import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import {EncounterDetails} from './EncounterDetails';
import * as EncounterInterface from '../../../../../encounter';

// Dummy array of encounters
import EncounterInstances from '../../../../../encounter_instances';

export interface IEncounterListState {
	viewEncounter: boolean,
	editEncounter: boolean,
	selectedEncounter: EncounterInterface.IEncounterState,
}

export class EncounterList extends React.Component<any, IEncounterListState> {
	constructor(props: any) {
		super(props);
		this.state = {
			viewEncounter: false,
			editEncounter: false,
			selectedEncounter: {} as EncounterInterface.IEncounterState,
		}
		this.resetState = this.resetState.bind(this);
	}

	resetState() {
		this.setState({ selectedEncounter: {} as EncounterInterface.IEncounterState});
		this.setState({ viewEncounter: false});
		this.setState({ editEncounter: false});
	}

	view = (encounter: EncounterInterface.IEncounterState) => {
		this.setState({ selectedEncounter: encounter});
		this.setState({ viewEncounter: true});
	}

	edit = (encounter: EncounterInterface.IEncounterState) => {
		this.setState({ selectedEncounter: encounter});
		this.setState({ editEncounter: true});
	}

	render() {
		if(!this.state.viewEncounter && !this.state.editEncounter){
			return (
				<div className= "layout card-grid">
					<Grid container spacing={40}>
						{EncounterInstances['EncounterInstances'].map(encounter => (
							<Grid item key={encounter.Id} sm={6} md={4} lg={3}>
								<Card className="card">
								  <CardMedia
									className="card-media"
									image="data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22288%22%20height%3D%22225%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20288%20225%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_164edaf95ee%20text%20%7B%20fill%3A%23eceeef%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A14pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_164edaf95ee%22%3E%3Crect%20width%3D%22288%22%20height%3D%22225%22%20fill%3D%22%2355595c%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%2296.32500076293945%22%20y%3D%22118.8%22%3EThumbnail%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E" // eslint-disable-line max-len
									title="Image title"
								  />
								  <CardContent className="card-content">
									<Typography gutterBottom variant="h5" component="h2">
									  {encounter.Name} {encounter.Id}
									</Typography>
									<Typography>
									  {encounter.Description}.
									</Typography>
								  </CardContent>
								  <CardActions>
									<Button size="small" color="primary" onClick={() => this.view(encounter)}>
									  View
									</Button>
									<Button size="small" color="primary" onClick={() => this.edit(encounter)}>
									  Edit
									</Button>
								  </CardActions>
								</Card>
							</Grid>
						))}
					</Grid>
				</div>
			)
		}
		else if(this.state.viewEncounter){
			return (<EncounterDetails encounter={this.state.selectedEncounter} resetParentState={this.resetState}/>);
		}
		else{
			// Todo: To enable editing, just use the same component as EncounterDetails:
			// enable edit button, remove readOnly, and add the code to submit the changes
			return (<div>Editing Feature is not yet enabled. Please check again later.</div>);
		}
	}
}
