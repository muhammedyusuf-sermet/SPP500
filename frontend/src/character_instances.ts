import { CharacterRace, CharacterClass, ICharacterData } from "./character";


let characterExample: ICharacterData;
characterExample = {
	Id: 1,
	Name: "Character",
	Level: 3,
	Race: CharacterRace.Gnome,
	Class: CharacterClass.Paladin,
	MaxHealth: 23,
	ArmorClass: 12,
	Notes: 'aggressive',

	Campaigns: []
};

export const CharacterInstances: ICharacterData[] = new Array();

// Generate a dummy array of characters with different IDs
for (var i = 0; i < 12; i++) {
	let character = {} as ICharacterData;
	Object.assign(character, characterExample);
	character.Id = i + 1;
	CharacterInstances.push(character);
}