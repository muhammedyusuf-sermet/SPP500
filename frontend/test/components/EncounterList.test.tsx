import * as React from "react"
import * as nock from 'nock';
import { shallow, ShallowWrapper } from 'enzyme';
import {EncounterList, IEncounterListState} from '../../src/renderer/components/platform/pages/view_game_components/EncounterList';
import * as EncounterInterface from '../../src/encounter';
import {API_URL} from '../../src/config'

jest.mock('../../src/cookie');

describe('Test the EncounterList View Details', () => {
	let encounterListInstance: ShallowWrapper<any, IEncounterListState, EncounterList>;

	beforeEach(() => {
		encounterListInstance = shallow(<EncounterList/>);
	})

	it('renders without crashing', () => {
		expect(encounterListInstance).toBeDefined();
	});

	it('renders correctly when the page is loaded', () => {
		expect(encounterListInstance).toMatchSnapshot();
	});

	it('should update the state variables when View button is clicked for a encounter', () => {
		let selectedEncounter: EncounterInterface.IEncounterState;
		selectedEncounter = {} as EncounterInterface.IEncounterState;
		encounterListInstance.instance().view(selectedEncounter);

		expect(encounterListInstance.state('viewEncounter')).toEqual(true);
		expect(encounterListInstance.state('selectedEncounter')).toEqual(selectedEncounter);
	});

	it('should update the state variables when Edit button is clicked for a encounter', () => {
		let selectedEncounter: EncounterInterface.IEncounterState;
		selectedEncounter = {} as EncounterInterface.IEncounterState;
		encounterListInstance.instance().edit(selectedEncounter);

		expect(encounterListInstance.state('editEncounter')).toEqual(true);
		expect(encounterListInstance.state('selectedEncounter')).toEqual(selectedEncounter);
	});

	it('should update the state variables when resetState function is called', () => {
		encounterListInstance.instance().resetState();

		expect(encounterListInstance.state('viewEncounter')).toEqual(false);
		expect(encounterListInstance.state('editEncounter')).toEqual(false);
		expect(encounterListInstance.state('selectedEncounter')).toEqual({} as EncounterInterface.IEncounterState);
	});

	it('should make an GET request to retrieve encounters when getPaginatedEncounters function is called', () => {
		encounterListInstance.instance().getPaginatedEncounters(0);

		nock(API_URL)
		.get('/encounter/get/0/12')
		.reply(201, { status: 201, message: 'success', total: 1, content: [] as EncounterInterface.IEncounterState[] });
	});

	it('should return the total number of pages when getTotalPages function is called', () => {
		encounterListInstance.instance().setState({ totalEncounters: 24});
		let totalPages = encounterListInstance.instance().getTotalPages();

		// Starts from 0
		expect(totalPages).toEqual(1);
	});
});
