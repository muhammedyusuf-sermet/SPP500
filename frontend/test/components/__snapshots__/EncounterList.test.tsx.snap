// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the EncounterList View Details renders correctly when the page is loaded 1`] = `
<div
  className="layout card-grid"
  id="view-encounters-container"
>
  <Pagination
    getTotalPages={[Function]}
    onPageChange={[Function]}
  />
  <WithStyles(Grid)
    container={true}
    spacing={40}
  />
</div>
`;
